core/string/ustring.cpp

```diff
@@ -1157,6 +1157,12 @@ Vector<String> String::split_spaces() const {
 
 Vector<String> String::split(const String &p_splitter, bool p_allow_empty, int p_maxsplit) const {
 	Vector<String> ret;
+
+	if (is_empty()) {
+		ret.push_back("");
```

Comment:
This should only be the case with allow_empty.

Reply:
Hm not sure. I just tested the following:
```
var split: PackedStringArray = "a b c".split("", false)
print(split)
```
Will print: `["a", " ", "b", " ", "c"]`

---
```
var split: PackedStringArray = "a b c".split("", true)
print(split)
```
Will also print: `["a", " ", "b", " ", "c"]`

So it looks like that `p_allow_empty` has no effect when the `p_splitter` is empty.

Reply:
Studied the code a bit more, can confirm my statement above.
An empty `p_splitter` behaves the same as `p_allow_empty` set to `true`.